### PR TITLE
feat(inspect): add Phase 5.2 reconciliation tab to the inspector overlay

### PR DESCRIPTION
**Pull request by danielraffel**
The reconciliation tab (R-key) answers one question per tweak: "will
this edit survive a re-import?" It is a read-only report over the
existing TweakStore + live view tree — no parallel data model. Every
stored tweak classifies into exactly one of three states:

  * locked-to-source — anchor is in the TweakStore lock set AND still
    resolves to a live view; the edit is promoted into authored source
    and survives a fresh re-import (the "reconciled" state).
  * drifted — anchor resolves to a live view but is not locked; the
    edit lives only in the runtime tweak layer (pulp-tweaks.json).
  * unresolvable — no live view carries the anchor (orphaned); the
    inspector cannot tell where the edit belongs, so it is shown as
    unresolvable rather than guessed or crashed on.

Builds on Phase 4a (lock-to-source) and Phase 5.1 (source-jump). The
tab toggles with R and takes over the property-panel region exactly
like the Phase 6.1 pass viewer; the pass viewer wins when both are on.

Adds 11 headless Catch2 cases to test/test_inspector.cpp covering the
R-key toggle, the three-state classification, the empty / no-store /
orphaned edge cases, and the lock-promotes-out-of-drift transition.
All 118 inspector tests pass.

Co-Authored-By: Claude Opus 4.7 (1M context) <noreply@anthropic.com>

### CODE REVIEW

File: inspect/src/inspector_overlay.cpp [hunk 1]
```diff
@@ -251,6 +256,78 @@ void InspectorOverlay::refresh_drift() {
...
+    std::unordered_set<std::string> live_anchors(live.begin(), live.end());
+
+    auto records = tweak_store_->list_tweaks();
+    // Stable ordering so the tab doesn't reshuffle across frames:
+    // group by anchor, anchors sorted, insertion order within an
+    // anchor (list_tweaks() preserves it).
+    std::sort(records.begin(), records.end(),
+              [](const TweakStore::Record& a, const TweakStore::Record& b) {
+                  return a.anchor_id < b.anchor_id;
+              });
```
[the remark this answers is not in the document]
Comment (danielraffel) [reply inspect/src/inspector_overlay.cpp:297]: Covered by follow-up #2546: tweak listing/reconcile ordering now preserves insertion order within an anchor, with focused TweakStore and reconcile-report regressions passing locally.
